src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -43,7 +50,10 @@ ReactRef.attachRefs = function(instance, element) {
   }
 };
 
-ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
+ReactRef.shouldUpdateRefs = function(
+  prevElement: ReactElement | null | false,
+  nextElement: ReactElement | null | false,
```

Comment:
Okay, this means that this code is behaving in a weird way. We are doing `.ref` of a number and string which doesn't make sense but happens to be working.